src/Microsoft.ML.TensorFlow/Microsoft.ML.TensorFlow.csproj

```diff
@@ -7,6 +7,11 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="System.IO.FileSystem.AccessControl" Version="4.5.0" />
+    <PackageReference Include="System.Security.Principal.Windows" Version="4.5.0" />
+  </ItemGroup>
```

Comment:
A) Do we really need them? Because presence of Principal.Windows, for multiplatform code is suspicious.
B) If we really need them you need to make them part of nuget dependency, and also we need to modify [https://github.com/dotnet/machinelearning/blob/master/build/Dependencies.props](https://github.com/dotnet/machinelearning/blob/master/build/Dependencies.props) instead of specify versions here.

Reply:
We use this as a security measure on windows platform. We are using temp directory for loading TFSession for SavedModels. For this temp directory we need to ACL it in the high-rights process so low-rights process can’t access.

I will look into  making it part of nuget dependency

---
In reply to: [219572600](https://github.com/dotnet/machinelearning/pull/853#discussion_r219572600) [](ancestors = 219572600)

Reply:
you can use this one as reference implementation [https://github.com/dotnet/machinelearning/blob/master/pkg/Microsoft.ML.ImageAnalytics/Microsoft.ML.ImageAnalytics.nupkgproj](https://github.com/dotnet/machinelearning/blob/master/pkg/Microsoft.ML.ImageAnalytics/Microsoft.ML.ImageAnalytics.nupkgproj)

---
In reply to: [219577320](https://github.com/dotnet/machinelearning/pull/853#discussion_r219577320) [](ancestors = 219577320,219572600)